src/etl/Migrations.cpp

```diff
@@ -0,0 +1,131 @@
+        auto const nft = backend->fetchFirstNFTState(id, yield);
+        NFTsData data = NFTsData(id, nft->ledgerSequence, nft->owner, {});
+        if (!nft->isBurned)
+        {
+            auto const bookmark = ripple::keylet::nftpage(
+                ripple::keylet::nftpage_min(nft->owner), id);
+            auto const last = ripple::keylet::nftpage_max(nft->owner);
+            ripple::uint256 nextKey = last.key;
+            std::optional<ripple::STLedgerEntry> sle;
+
+            do
+            {
+                auto const blob = backend->fetchLedgerObject(
+                    ripple::Keylet(ripple::ltNFTOKEN_PAGE, nextKey).key,
+                    nft->ledgerSequence,
+                    yield);
+
+                if (!blob || blob->size() == 0)
+                {
+                    std::stringstream ss;
+                    ss << "Error with migration " << __func__ << ": " << id
+                       << " cannot find NFTokenPage";
+                    throw std::runtime_error(ss.str());
+                }
+
+                sle = ripple::STLedgerEntry(
+                    ripple::SerialIter{blob->data(), blob->size()}, nextKey);
+                if (sle->isFieldPresent(ripple::sfPreviousPageMin))
+                    nextKey = sle->getFieldH256(ripple::sfPreviousPageMin);
+            } while (sle && sle->key() != nextKey && nextKey > bookmark.key);
+
+            if (!sle)
+            {
+                std::stringstream ss;
+                ss << "Error with migration " << __func__ << ": " << id
+                   << " cannot find NFTokenPage";
+                throw std::runtime_error(ss.str());
+            }
+
+            auto const ledgerNfts = sle->getFieldArray(ripple::sfNFTokens);
+            auto const ledgerNft = std::find_if(
+                ledgerNfts.begin(),
+                ledgerNfts.end(),
+                [&id](ripple::STObject const& candidate) {
+                    return candidate.getFieldH256(ripple::sfNFTokenID) == id;
+                });
+
+            if (ledgerNft == ledgerNfts.end())
+            {
+                std::stringstream ss;
+                ss << "Error with migration " << __func__ << ": " << id
+                   << " cannot find NFTokenPage";
+                throw std::runtime_error(ss.str());
+            }
+
+            data.uri = ledgerNft->getFieldVL(ripple::sfURI);
+            toWrite.push_back(data);
+            continue;
+        }
+
+        // 3 - otherwise if we have no record of it not burned we cannot
+        // determine the uri but we can still write the taxon into the issuer
+        // table
+        std::stringstream ss;
+        ss << "This shouldn't happen: " << id;
+        throw std::runtime_error(ss.str());
+        // data.isBurned = true;
+        // toWrite.push_back(data);
+    }
+
+    // write the new data
+    backend->writeNFTs(std::move(toWrite));
+
+    // write the migration receipt
+    backend->writeMigrationReceipt(__func__);
```

Comment:
Please make sure that `__func__` is returns the same string on all supported compilers (it should, but better check).